From 5122f7925b15f23d3975547e6fd9a3e1b677e1c7 Mon Sep 17 00:00:00 2001
From: "Thulasimani,Sivakumar" <sivakumar.thulasimani@intel.com>
Date: Tue, 12 Jan 2016 11:08:37 +0530
Subject: [PATCH 5/5] FOR_UPSTREAM [VPG]: drm/i915: Add vco values for CHT DP

This patch adds VCO values to the precalculated pll values
for both link rates used with DP panels. These were
generated with the chv_find_best_dpll logic and
hardcoded since DP can use a limited number of
link rates only, which in CHT happens to be 2.
HBR2 has been removed from the list as it is not
supported on CHT.

Change-Id: I325eb61764765f59b078cefa48af8894f51eab3e
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-14849
Signed-off-by: Sivakumar Thulasimani <sivakumar.thulasimani@intel.com>
Reviewed-on: https://android.intel.com:443/468194
---
 drivers/gpu/drm/i915/intel_dp.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_dp.c b/drivers/gpu/drm/i915/intel_dp.c
index e72349b..993f53d 100644
--- a/drivers/gpu/drm/i915/intel_dp.c
+++ b/drivers/gpu/drm/i915/intel_dp.c
@@ -102,11 +102,11 @@ static const struct dp_link_dpll chv_dpll[] = {
 	 * (m2_int << 22) | m2_fraction
 	 */
 	{ DP_LINK_BW_1_62,	/* m2_int = 32, m2_fraction = 1677722 */
-		{ .p1 = 4, .p2 = 2, .n = 1, .m1 = 2, .m2 = 0x819999a } },
+		{ .p1 = 4, .p2 = 2, .n = 1, .m1 = 2, .m2 = 0x819999a,
+			.vco = 0x62e080 } },
 	{ DP_LINK_BW_2_7,	/* m2_int = 27, m2_fraction = 0 */
-		{ .p1 = 4, .p2 = 1, .n = 1, .m1 = 2, .m2 = 0x6c00000 } },
-	{ DP_LINK_BW_5_4,	/* m2_int = 27, m2_fraction = 0 */
-		{ .p1 = 2, .p2 = 1, .n = 1, .m1 = 2, .m2 = 0x6c00000 } }
+		{ .p1 = 4, .p2 = 1, .n = 1, .m1 = 2, .m2 = 0x6c00000,
+			.vco = 0x5265c0 } },
 };
 
 /**
-- 
1.9.1

